### PR TITLE
[NNAPI EP] Move GetCapability independent of ModelBuilder

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: [NNAPI EP] Move GetCapability independent of ModelBuilder

**Motivation and Context**
- This is the 2nd step to enable NNAPI EP GetCapability without actually running on Android, the previous PR is #5746
- Move GetSupportedNodes and IsNodeSupported from ModelBuilder to helper
- Make OpBuilders and OpSupportCheckers static global 
- Minor code and style update

### CODE REVIEW

File: onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc [hunk 1]
```diff
@@ -53,7 +54,11 @@ NnapiExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_view
   }
 
   nnapi::ModelBuilder builder(graph_view);
-  const auto supported_nodes_vector = builder.GetSupportedNodes();
+  nnapi::OpSupportCheckParams params{
+      builder.GetAndroidSdkVer(),
```
Comment (gwang-msft) [inline onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc:58]: This is the only part I cannot get rid of ModelBuilder, how do we pass the target_android_sdk_ver to NNAPI EP?

Add the target_android_sdk_ver to NNAPI EP ctor
Use session config options
Compile def?
Will address this in next PR

File: onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc [hunk 2]
```diff
@@ -535,7 +536,12 @@ class BaseOpBuilder : public IOpBuilder {
 };
 
 Status BaseOpBuilder::AddToModelBuilder(ModelBuilder& model_builder, const Node& node) const {
-  ORT_RETURN_IF_NOT(model_builder.IsNodeSupported(node), "Unsupported operator ", node.OpType());
+  OpSupportCheckParams params{
+      model_builder.GetAndroidSdkVer(),
+      model_builder.UseNCHW(),
+  };
+
```
Comment (skottmckay) [inline onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc:543]: Do we need to construct this on every call to AddToModelBuilder?

**Review by skottmckay — APPROVED** [review]
:shipit:

File: onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc (hunk 2 above)
Comment (gwang-msft) [reply onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc:543]: Can create one and store it in ModelBuilder, will address this in next PR

**Pull request merged**